src/graphnet/components/loss_functions.py

```diff
@@ -45,15 +47,19 @@ def forward(
             Tensor: Loss, either averaged to a scalar (if `return_elements = False`)
                 or elementwise terms with shape [N,] (if `return_elements = True`).
         """
-        elements = self._forward(prediction, target)
+        elements = self._forward(prediction, target)  # , weights)
+        if weights is not None:
+            elements = elements * weights
         assert elements.size(dim=0) == target.size(
             dim=0
         ), "`_forward` should return elementwise loss terms."
 
         return elements if return_elements else torch.mean(elements)
 
     @abstractmethod
-    def _forward(self, prediction: Tensor, target: Tensor) -> Tensor:
+    def _forward(
+        self, prediction: Tensor, target: Tensor, weights: Tensor
+    ) -> Tensor:
```

Comment:
I suppose this should be reverted given the rewrite?
```suggestion
    def _forward(self, prediction: Tensor, target: Tensor) -> Tensor:
```